In [1]:
import numpy as np
import scipy.stats as stats

In [2]:
data=np.array([[18,13,12,15],
               [24,6,6,6]])

In [3]:
(chi2,p,dof,E)=stats.chi2_contingency(data,correction=False)

In [7]:
# (b) The expected table under H0:
E

array([[24.36, 11.02, 10.44, 12.18],
       [17.64,  7.98,  7.56,  8.82]])

In [8]:
# (c) The errors table:
data-E

array([[-6.36,  1.98,  1.56,  2.82],
       [ 6.36, -1.98, -1.56, -2.82]])

In [10]:
# (d) The chisquared statistic:
chi2

6.910132474042249

In [16]:
# Check:
np.sum((data-E)**2/E)

6.910132474042249

In [17]:
# (e) The number of degrees of freedom:
dof

3

In [20]:
# Check:
size=np.shape(data)
size

(2, 4)

In [21]:
(size[0]-1)*(size[1]-1)

3

In [22]:
# (f) p-value:
p

0.07481799078534634

(g) No, the $p$-value for this table is 0.075, which is bigger than the specified $p$-value threshold